README.md

```diff
@@ -31,3 +31,29 @@ Checks for major and minor version to match, i.e. installed **1.0.1** and docume
 ./vendor/bin/depdoc validate --very-strict
 ```
 Checks for full semantic versioning match, i.e. installed **1.0.1** and documented **1.0.0** will not bypass validation.
+
+# Configuration
+
+Configuration can be provided via _.depdoc.json, .depdoc.yml or .depdoc.yml_ files in the root directory of your project.
+
+## Options
+
+**newlineCharacter**
+
+Default is _\n_ but if you need a different one you can provide it.
+
+**lockSymbol**
+
+Default is "🔒". Alternatively you can use 🛇, ⚠, or ✋.
+
+**exportExternalLink**
+
+Default is _true_, but if you do not want the external link in your dependencies file you can deactivate it.
+
+**composer**
+
+Default is _true_, if you do not want to add composer dependencies set this to _false_
+
+**npm**
+
+Default is _true_, if you do not want to add npm dependencies set this to _false_.
```

Comment:
Okay, sorry, das muss ich hier mal gerade in Deutsch machen, weil es mich zu sehr verwirrt:
Stehe ich gerade komplett auf dem Schlauch oder sind Teile unserer Config komplett überflüssig/ungenutzt? 😅 Die `ApplicationConfiguration` wird im `BaseCommand` ausgelesen (oder mit Defaults erzeugt). In den Commands kann man diese Config dann auslesen und im `execute` für sonst was verwenden. So wie du es eben nun auch im `BaseCommand` machst.
Die einzige andere Stelle, wo dies geschieht, ist das Setzen der New-Line in der `WriteConfiguration`. `exportExternalLink` wird dabei derzeit noch ignoriert.
Für ``lockSymbol`` gibt - soweit ich das sehe - aktuell nicht mal eine Möglichkeit, dies einfließen zu lassen. Beim Auslesen einer bestehenden Markdown-Datei wird das jeweils verwendete Lock-Symbol in ``DependencyData::$lockSymbol`` zwischengespeichert und später wieder zurückgeschrieben. 😬 Ich könnte mir aber mal ein Ticket anlegen, um ebendies zu korrigieren.

Nun aber noch zur eigentlichen Erweiterung: Die finde ich top! Danke für's Umsetzen.

Reply:
Japs du hast Recht, die halbe Config ist überflüssig.

Reply:
Also am besten diesen PR erstmal mergen und über ein weiteren Issue die Lock-Symbol-Option, sowie die Option, ob mit oder ohne External Link dann berücksichtigen, nicht wahr?
Das mache ich dann auch gerne mal wieder :-)